### PR TITLE
Add iverilog recipe

### PR DESCRIPTION
**Pull request by tfors**

### CODE REVIEW

File: recipes/gperf/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+./configure --prefix=$PREFIX
+make
```
Comment (jakirkham) [inline recipes/gperf/build.sh]: Please add `make check` or similar to run tests.

File: recipes/iverilog/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+package:
+  name: iverilog
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/steveicarus/iverilog.git
+  git_tag: {{ git_tag }}
+  patches:
+    - remove_static_build_path.patch
```
Comment (jakirkham) [inline recipes/iverilog/meta.yaml]: Has this patch been sent upstream?

File: recipes/iverilog/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{% set version = "10.1" %}
+{% set git_tag = "v10_1" %}
+
+package:
+  name: iverilog
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/steveicarus/iverilog.git
+  git_tag: {{ git_tag }}
```
Comment (jakirkham) [inline recipes/iverilog/meta.yaml]: Please change this to download a tarball instead.

Comment (jakirkham) [inline recipes/iverilog/meta.yaml]: The `url` should then be `https://github.com/steveicarus/iverilog/archive/{{ git_tag }}.tar.gz`.

File: recipes/iverilog/meta.yaml (hunk 2 above)
Comment (tfors) [inline recipes/iverilog/meta.yaml]: Not yet, but I will.

Comment (jakirkham) [inline recipes/iverilog/meta.yaml]: Please reference the PR in a comment above this patch so that it will be easy to track when we can drop it.

**Comment by jakirkham** [comment]
Windows failure was a [fluke]( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.4673/job/pf4ili7cj0df66mq ). Restarted...

**Comment by jakirkham** [comment]
Though I guess they all skip Windows.

File: recipes/gperf/build.sh (hunk 1 above)
Comment (jakirkham) [inline recipes/gperf/build.sh]: Done below.

File: recipes/sed/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  url: http://ftp.gnu.org/gnu/sed/sed-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win]
+
+# requirements:
+#   build:
+#     - sed
```
Comment (jakirkham) [inline recipes/sed/meta.yaml]: Can we please drop this?

File: recipes/gperf/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  url: http://ftp.gnu.org/pub/gnu/gperf/gperf-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - sed  # [osx]
```
Comment (jakirkham) [inline recipes/gperf/meta.yaml]: So, OS X already has `sed`. Does that not work?

File: recipes/iverilog/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+package:
+  name: iverilog
+  version: {{ version }}
+
+source:
+  fn: {{ git_tag }}.tar.gz
+  url: https://github.com/steveicarus/iverilog/archive/{{ git_tag }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - remove_static_build_path.patch
```
Comment (jakirkham) [inline recipes/iverilog/meta.yaml]: Please provide a comment describing the patch and the PR it appears in.

File: recipes/gperf/meta.yaml (hunk 5 above)
Comment (tfors) [inline recipes/gperf/meta.yaml]: Unfortunately, the gperf tests fail using OSX's BSD sed with:

```
../src/gperf -C -E -G -I -t < ./languages.gperf | LC_ALL=C tr -d '\r' | sed -e 's|[^ ]*[/\\][\\]*src[/\\][\\]*gperf[^ ]*|../src/gperf|' > languages.out
sed: RE error: illegal byte sequence
make[1]: *** [check-test] Error 1
make: *** [check] Error 2
```
Using gnu sed, the tests pass.

Comment (jakirkham) [inline recipes/gperf/meta.yaml]: Yeah Mac's `sed` is a bit quirky. Seems reasonable.

File: recipes/iverilog/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+source:
+  fn: {{ git_tag }}.tar.gz
+  url: https://github.com/steveicarus/iverilog/archive/{{ git_tag }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    # Submitted upstream: PR: steveicarus/iverilog#116
+    # Uses /proc/self/exe at run-time to find ivl_root instead of
+    # static path at build-time.
+    - remove_static_build_path.patch
```
Comment (jakirkham) [inline recipes/iverilog/meta.yaml]: 👍 Thanks for doing this.

xref: https://github.com/steveicarus/iverilog/pull/116

File: recipes/sed/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/sed/meta.yaml]: I think this is the only thing left.

**Comment by jakirkham** [comment]
There is one lingering comment that needs to be snipped. Once that is done I'm happy with it.

File: recipes/iverilog/meta.yaml (hunk 7 above)
Comment (tfors) [inline recipes/iverilog/meta.yaml]: No problem.  Thanks for all the feedback!

**Comment by jakirkham** [comment]
Thanks @tfors.

**Comment by jakirkham** [comment]
Soon you should get an invitation to join conda-forge. Once accepted, you will be added to a team for each recipe. Each team will have permissions on a feedstock with that recipe. When you wish to make changes, please follow the typical GitHub model.

 1. Fork the feedstock.
 2. Make changes to your fork.
 3. PR them back to the feedstock.

If you have any questions, please let us know. Welcome to conda-forge. 😄

**Comment by jakirkham** [comment]
I forgot to mention the `toolchain`. 😩 Anyways, I proposed a PR for `gperf`, which should be an easy merge. Also there is one for `iverilog`, but we can get to that after `gperf` is done.

**Comment by jakirkham** [comment]
@steveicarus @caryr @orsonmmz @martinwhitaker @ldoolitt, just letting you know we have added `iverilog` to [conda-forge]( http://conda-forge.github.io/ ). This will allow users to easily install this with [`conda`]( http://conda.pydata.org/docs/ ) (`conda install -c conda-forge iverilog`). The maintenance burden here is light as all testing releases are built and deployed automatically with CI. Changes to how it is being built can be controlled by changing the recipe in [this repo]( https://github.com/conda-forge/iverilog-feedstock ). If you have any interest in being added as a maintainer (to make new releases for instance), we would love to have you on board. Thanks for your work on `iverilog`. 😄

**Pull request merged**